OCPBUGS 4543: Prioritizing deployments over deployment configs (#63036)
openshift/openshift-docs: modules/deployments-replicationcontrollers.adoc

The replication controller does not perform auto-scaling based on load or traffic, as it does not track either. Rather, this requires its replica
count to be adjusted by an external auto-scaler.

[NOTE]
====

> It is recommended to use a `DeploymentConfig` to create a replication controller. It is not recommended to create replication controllers directly.

Reviewer: +1, and maybe we can add they should use replica sets if they want to create them directly

Author: @atiratree can you explain this a bit more? If you want to create <what> directly? (Because creating a replica set wouldn't create a replication controller I assume)

(And is it an okay idea to create replica sets directly, or is that not something we should actually recommend?)

Reviewer: as we mention in the replica note something in this regard:

If you require custom update orchestration or do not require updates at all, use replica sets instead of replication controllers.

The replication controller does not perform auto-scaling based on load or traffic, as it does not track either. Rather, this requires its replica
count to be adjusted by an external auto-scaler.

[NOTE]
====

> It is recommended to use a `DeploymentConfig` to create a replication controller. It is not recommended to create replication controllers directly.

Reviewer 2: Suggestion to avoid the word "recommend", which the [RHSSG recommends (heh) we avoid](https://redhat-documentation.github.io/supplementary-style-guide/#recommend):

```suggestion
Use a `DeploymentConfig` to create a replication controller. Do not create replication controllers directly.
```

Author: Thanks, reworked it some. This guidance is one that I always want to break because I sometimes feel like the "do" and "do not" are too harsh. But I think I reworked it so I'm happy with it now
